contracts/v2/sovereignChains/BridgeL2SovereignChain.sol

```diff
@@ -0,0 +1,393 @@
+            if (_sovereignWETHAddress == address(0)) {
+                // Create a wrapped token for WETH, with salt == 0
+                WETHToken = _deployWrappedToken(
+                    0, // salt
+                    abi.encode("Wrapped Ether", "WETH", 18)
+                );
+            } else {
+                WETHToken = TokenWrapped(_sovereignWETHAddress);
+                wrappedAddressIsNotMintable[
+                    _sovereignWETHAddress
+                ] = _sovereignWETHAddressIsNotMintable;
+            }
+        }
+
+        // Initialize OZ contracts
+        __ReentrancyGuard_init();
+    }
+
+    modifier onlyBridgeManager() {
+        if (bridgeManager != msg.sender) {
+            revert OnlyBridgeManager();
+        }
+        _;
+    }
+
+    /**
+     * @notice Updated bridge manager address
+     * @param _bridgeManager Bridge manager address
+     */
+    function setBridgeManager(
+        address _bridgeManager
+    ) external onlyBridgeManager {
+        if (_bridgeManager == address(0)) revert NotValidBridgeManager();
+        bridgeManager = _bridgeManager;
+        emit BridgeManagerUpdated(bridgeManager);
+    }
+
+    /**
+     * @notice Remap multiple wrapped tokens to a new sovereign token address
+     * @dev This function is a "multi/batch call" to `setSovereignTokenAddress`
+     * @param sovereignTokenAddresses Array of SovereignTokenAddress to remap
+     */
+    function setMultipleSovereignTokenAddress(
+        SovereignTokenAddress[] calldata sovereignTokenAddresses
+    ) external onlyBridgeManager {
+        // Make multiple calls to setSovereignTokenAddress
+        for (uint256 i = 0; i < sovereignTokenAddresses.length; i++) {
+            _setSovereignTokenAddress(
+                sovereignTokenAddresses[i].originNetwork,
+                sovereignTokenAddresses[i].originTokenAddress,
+                sovereignTokenAddresses[i].sovereignTokenAddress,
+                sovereignTokenAddresses[i].isNotMintable
+            );
+        }
+    }
+
+    /**
+     * @notice Remap a wrapped token to a new sovereign token address
+     * @dev This function is used to allow any existing token to be mapped with
+     *      origin token.
+     * @notice If this function is called multiple times for the same existingTokenAddress,
+     * this will override the previous calls and only keep the last sovereignTokenAddress.
+     * @notice The tokenInfoToWrappedToken mapping  value is replaced by the new sovereign address but it's not the case for the wrappedTokenToTokenInfo map where the value is added, this way user will always be able to withdraw their tokens
+     * @param originNetwork Origin network
+     * @param originTokenAddress Origin token address, 0 address is reserved for ether
+     * @param sovereignTokenAddress Address of the sovereign wrapped token
+     * @param isNotMintable Flag to indicate if the wrapped token is not mintable
+     */
+    function setSovereignTokenAddress(
```

Comment:
i think the sufix address here might be confusing

setSovereignToken?¿

Reply:
same with other functions/evnets htat has de sufix address

Reply:
Mhhh in my opinion it is suitable as we are mainly updating the address of the mapped token (and the mintable flag)